pygnmi/client.py

```diff
@@ -673,20 +680,35 @@ def set(
                 )
 
             if gnmi_extension:
-                gnmi_message_request = SetRequest(
-                    prefix=protobuf_prefix,
-                    delete=del_protobuf_paths,
-                    update=update_msg,
-                    replace=replace_msg,
-                    extension=[gnmi_extension],
-                )
+                if prefix is None and target is None:
```

Comment:
Something in the lines of below will be a create once and use everywhere mode. 

```py
# Define the base arguments for the SetRequest
gnmi_message_request_args = {
    'delete': del_protobuf_paths,
    'update': update_msg,
    'replace': replace_msg,
    'extension': [gnmi_extension]
}

# Add the prefix only if it's necessary
if gnmi_extension and (prefix is not None or target is not None):
    gnmi_message_request_args['prefix'] = protobuf_prefix

# Create the SetRequest with the specified arguments
gnmi_message_request = SetRequest(**gnmi_message_request_args)

```

Reply:
I agree, I was only staying true to the current coding convention in the codebase.